index.js

```diff
@@ -85,12 +85,31 @@ app.post("/upload", async (req, res, next)=>{
 						
 	busboy.on("file", async (field, file, name) => {
 		console.log(req.body, name)
-		const target = await drive.make(req.query.id,  name, false, { preventNameCollision: true });
-		res.send(await drive.info(await drive.write(target, file)));
+
+		busboy.on('field', async function(field, val) {
```

Comment:
this is unreliable approach

- code depends on order of data in the post request ( if field was provided before file, handler will not be triggered ). To make it more stable we need two handler defined on the same level, which will store the file/path info and "finish" handler in which we will do all processing. 

- field gathering logic assumes that there will be only one filed, which is not correct. We need to check the field name at least

Reply:
2. yes, thanks, I will fix this and check the name

1. writing file in "finish" is problematic, though... 
 
according to the author of busboy, when file is received in "file", it has to be consumed somehow. otherwise the request will just hang and "finish" will never be called. 'consuming' that allows to move on is e.g. `file.resume()`, `file.on("data", chunk => { ... })` or writing it somewhere (`drive.write(path, file)`, `file.pipe(someWritableStream)`. 

a. I tried the first 2, but on "finish" the readable stream is not readable anymore after this and the files are created, but with no content, and requests hang.
![Screenshot from 2021-11-19 12-49-41](https://user-images.githubusercontent.com/31205423/142620737-8ca74181-eb7f-4048-ba6a-64acf436ce6c.png)

 b. I also tried a somewhat ugly approach that almost works:
- write files in req.query.id (upload root)
- then in "finish" check if there is a folder for those files, create it if necessary and move them in that folder. this works but for one case: when we have at least two folders with same name files like
  - folder a: file1.txt
  - folder b: file1.txt
in this case both files are written to root by the same name, and since they are written there at approximately the same time, only 1 file is really created

I think I am missing something, but for now this is an impasse :(

Reply:
the best approach would be to send file name in url query instead of the post data, but it will require unnecessary complication of client side code.
so, I will accept the pull request as is